gno.land/pkg/gnoweb/webclient_html.go

```diff
@@ -145,6 +147,11 @@ func (s *HTMLWebClient) SourceFile(w io.Writer, path, fileName string, isRaw boo
 		SizeKb: float64(len(source)) / 1024.0,
 	}
 
+	// If writer is nil, just return metadata
+	if w == nil {
+		return &fileMeta, nil
+	}
+
```

Comment:
writer cannot and should not be nil; remove this. it will complexity the caller logic